### PR TITLE
#161966916 Implement Article CRUD

### PR DESCRIPTION
**Pull request by rkemmy**
### What does this PR do?
Implement the article CRUD

### Description of Task to be completed
Implement endpoints for
* Creating an article 
* Getting all articles 
* Retrieving a specific article by id
* Update a specific article
* Delete a specific article
* Adding slug functionality
* Configure the code to pep8 standards
* Refactor tests

### How should this be manually tested?
* `git clone -b ft-create-article-161966916 https://github.com/andela/ah-infinity-stones` and ` cd ah-infinity-stones`.
* Run `python3 -m venv virtual` to create a virtual environment and `source virtual/bin/activate` to activate it.
* `pip install -r requirements.txt` to install dependencies
* Create a test database in the postgres shell
* Set up environment variables using the sample below as a guide:
```export DATABASE_USER='<postgres username>'
export DATABASE_PASSWORD='<postgres user's password>'
export HOST='localhost'
export PORT='5432'
export DATABASE_NAME='<name_of_database>' 
export SECRET_KEY='<personal_secret_key>'
export EMAIL_HOST_USER='<email_host_username>'
export EMAIL_HOST_PASSWORD='<email_host_password>'
```
* Run `python3 manage.py makemigrations` followed by `python3 manage.py migrate` to apply migrations
 Acess `http://127.0.0.1:8000/api/users/` through [Postman](https://www.getpostman.com/), provide inputs in the following format and create account.
```
 {
	"user": {
		"username": "remmykn",
		"email": "remmy@test.com",
		"password": "Password@123"
	}
}

```
Check your email and activate the account. Use the registration token to access the following endpoints:
- `http://127.0.0.1:8000/api/articles` - Post an article and get all articles
-`http://127.0.0.1:8000/api/articles/<slug>` - To get a specific article to either update or delete.
*  Run tests`python3 manage.py test authors.apps.articles.tests.test_views`

### Relevant pivotal tracker stories
[Delivers #161966916](https://www.pivotaltracker.com/story/show/161966916)

### Screenshots
image

image

image

image

image

image

### CODE REVIEW

**Comment by codeplus254** [comment]
Hey @rkemmy you could change the key 'user' in the article response to 'author' to make it more descriptive. Great work!

**Comment by rkemmy** [comment]
> Hey @rkemmy you could change the key 'user' in the article response to 'author' to make it more descriptive. Great work!
Thanks for pointing that out. I made the change.

File: Procfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+release: ./release_file.sh
```
Comment (r-wambui) [inline Procfile]: @rkemmy do we need this?

**Comment by r-wambui** [comment]
@rkemmy Workon squashing commit, the commit we have currently is not related to the PR.

**Comment by r-wambui** [comment]
@rkemmy rebase develop  and ensure that the tests on Articles CRUD are passing on travis.

File: Procfile (hunk 1 above)
Comment (rkemmy) [reply Procfile]: Sorry, I think the changes that Leo made are still reflected. I will remove that

**Comment by rkemmy** [comment]
@r-wambui I edited Leo's commit. I think that's why that happened but I will create a new commit.

**Pull request merged**